### PR TITLE
fix: Prevent AutoHeight container from triggering update just before unMounting

### LINKED ISSUE #20083: [Task]: Prevent AutoHeight container from triggering update just before unMounting
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

This task is to prevent AutoHeightContainer from triggering Auto Height Update just before un-mounting of the component

### PR DESCRIPTION
**Pull request by rahulramesha**
## Description

This PR includes the following changes,

- Changes to make sure the AutoHeightContainer does trigger Action while the component is unMounting
- A final check while Auto Height is still enabled by the time the updates for the widget is processed

Fixes #20083 

## Type of change

- Bug fix (non-breaking change which fixes an issue),(This fixes an issue in the upcoming List Widget V2 PR)
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?
Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/sagas/autoHeightSagas/widgets.ts [hunk 1]
```diff
@@ -93,7 +93,11 @@ export function* updateWidgetAutoHeightSaga() {
     // Get the widget from the reducer.
     const widget: FlattenedWidgetProps = stateWidgets[widgetId];
     // If this widget exists (not sure why this needs to be here)
-    if (widget && !widget.detachFromLayout) {
+    if (
+      widget &&
+      !widget.detachFromLayout &&
+      isAutoHeightEnabledForWidget(widget)
```
Comment (riodeuno) [inline app/client/src/sagas/autoHeightSagas/widgets.ts:99]: This is quite odd. Usually, updates don't come for widgets which don't have auto height.

**Review by riodeuno — APPROVED** [review]
Looks fine to me!

File: app/client/src/components/autoHeight/AutoHeightContainer.tsx [hunk 2]
```diff
@@ -50,9 +59,9 @@ export default function AutoHeightContainer({
     }
 
     return () => {
-      if (ref.current) {
-        observer.current.unobserve(ref.current);
-      }
+      // clearing out timeout if the component is unMounting
+      unmountingTimeout.current && clearTimeout(unmountingTimeout.current);
+      observer.current.disconnect();
```
Comment (rahulramesha) [inline app/client/src/components/autoHeight/AutoHeightContainer.tsx:64]: Added this so to unobserve all, even when ref.current is null, which seemed to be the case while un mounting.

**Review by riodeuno — APPROVED** [review]

**Pull request merged**